w04-associations-and-auth/d3-homework/README.md

```diff
@@ -0,0 +1,150 @@
+
+See? You did it!
+
+**Example Scope**
+
+My app (using the Google Maps API) will allow users to add markers to a map, displaying a friend's name and city in a sidebar. My stretch goal is to move the sidebar info into individual info-windows on each marker that display when clicked.
+
+To help break out your scope, you'll want milestones.
+
+### Milestones
+
+Milestones provide checkpoints for the progress of your app – they give you a way to check in with yourself about how much headway you've made.
+
+Listing each feature reminds you what is done or pending, and makes it convenient for estimating the number of hours/days for a feature.
+
+*Note: each milestone will likely be made of several user stories, and contain elements of the technical requirements.*
+
+**Example Milestones**
+
+- [ ] project planning complete (wireframes, user stories, etc.)
+- [ ] Hello World achieved
+- [ ] Save and render data from database using a form
+- [ ] authentication (sign up, log in, logout) working
+- [ ] API call to <External API HERE> being saved in/served from database
+- [ ] client styling (CSS) complete
+- [ ] app deployed to Heroku
+
+### Feasibility
+
+Before you get started, you'll want to do some research to see if what you're looking to do is possible. Some areas that you might want to investigate, depending on your app's desired functionality are:
+
+- Reading documentation to see what data you can request from your API of choice. Is it JSON? XML? Is all the information you want included in the response? Will you need to make several different requests to the API?
+- Verifying that you can successfully request data from your API using [Postman](https://chrome.google.com/webstore/detail/postman-rest-client/fdmmgilgnpjigdojojpjoooidkmcomcm?hl=en) or [cURL](http://conqueringthecommandline.com/book/curl#cid23)
+- Researching something you want to use that hasn't been taught in class yet, like an external module/library, data store, etc.
+
+**Example Feasibility**
+
+- [ ] Read Yelp API documentation
+- [ ] Use Postman to test Twilio API
+- [ ] Research using Handlebars.js client-side templating
+
+
+## Outside-in Development
+
+Outside-in development says you should start with the "outside" of your app - the views that the user sees, and move backwards to the server, then the database. Don't try to develop the whole front end of your app before moving on, but for each user story, page, or feature, you can follow the outside-in pattern.
+
+Start with the basics of your view:
+
+- [ ] an index.html file with static data directly in the file
+- [ ] create a template based on the structure in the html
+- [ ] use the template to display dynamic data stored on the client
```

Comment:
I don't get what this means...  How are we going to display dynamic data stored on the client if it's a server side template?